### PR TITLE
Allow lookups to be used on dimensions in filters

### PR DESCRIPTION
**Pull request by legoscia**

### CODE REVIEW

File: lib/panoramix/query.ex [hunk 1]
```diff
@@ -436,19 +442,40 @@ defmodule Panoramix.Query do
...
+    %{dimension: dimension}
   end
-  defp maybe_build_dimension(_) do
+  defp dimension_or_extraction_fn({:lookup, _, args}) do
+    case args do
+      [from_dimension, lookup_name] ->
+        %{dimension: Atom.to_string(from_dimension),
+          extractionFn: {:%{}, [],
+                         [{"type", "registeredLookup"},
+                          {"lookup", lookup_name}]}}
```
Comment (maxkuzmins) [inline lib/panoramix/query.ex:460]: Not sure how critical it is for a typical use case, but it looks like there's no way to pass a property for registered lookup (e.g. `retainMissingValue`, `replaceMissingValueWith`, `injective`, `optimize`). 
Perhaps, that can be done by allowing a list of properties to be appended in the `args`. E.g.

```
    case args do
      [from_dimension, lookup_name, opts] ->
        %{dimension: Atom.to_string(from_dimension),
          extractionFn: {:%{}, [],
                         [{"type", "registeredLookup"},
                          {"lookup", lookup_name}] ++ opts}}
    ...
```

Comment (legoscia) [reply lib/panoramix/query.ex:460]: That's a good point, I'll add that.

File: lib/panoramix/query.ex [hunk 2]
```diff
@@ -453,11 +453,15 @@ defmodule Panoramix.Query do
   end
   defp dimension_or_extraction_fn({:lookup, _, args}) do
     case args do
-      [from_dimension, lookup_name] ->
+      [from_dimension, lookup_name | maybe_opts] ->
+        opts = case maybe_opts do
+                 [] -> []
+                 [opts] -> opts
```
Comment (maxkuzmins) [inline lib/panoramix/query.ex:459]: Would that work if `maybe_opts` contains more than 1 option in a list?

**Comment by legoscia** [comment]
I decided to change the syntax, to handle the case where we need to chain one lookup to another.

File: lib/panoramix/query.ex (hunk 2 above)
Comment (legoscia) [reply lib/panoramix/query.ex:459]: Yes, I added a test that exercises that.

**Pull request merged**